### PR TITLE
fix(vm): cpu usage/idle handled by single worker

### PR DESCRIPTION
**Pull request by JimMoen**
Fixes [https://emqx.atlassian.net/browse/EMQX-11897](https://emqx.atlassian.net/browse/EMQX-11897)

Release version: v/e5.6.0

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [x] Added property-based tests for code which performs user input validation
- [x] Changed lines covered in coverage report
- [x] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [x] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [x] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- ~[ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)~
- ~[ ] Change log has been added to `changes/` dir for user-facing artifacts update~

### CODE REVIEW

File: apps/emqx/src/emqx_sys_sup.erl [hunk 1]
```diff
@@ -28,7 +28,8 @@ start_link() ->
 init([]) ->
     OsMon =
         case emqx_os_mon:is_os_check_supported() of
-            true -> [child_spec(emqx_os_mon)];
+            %% emqx_cpu_sup_worker is required by emqx_os_mon
```
Comment (thalesmg) [inline apps/emqx/src/emqx_sys_sup.erl:31]: If `emqx_cpu_sup_worker` is required by `emqx_os_mon` and this supervisor has `one_for_one` restart strategy, maybe it would be safer to spawn these 2 processes under a new supervisor that uses `rest_for_one` strategy.

File: apps/emqx_management/src/emqx_mgmt.erl [hunk 2]
```diff
@@ -205,23 +205,17 @@ cpu_stats() ->
...
-        _ -> 0
-    end;
-vm_stats('cpu.use') ->
-    case vm_stats('cpu.idle') of
-        0 -> 0;
-        Idle -> 100 - Idle
+vm_stats('cpu') ->
+    CpuUtilArg = [],
+    case emqx_vm:cpu_util([CpuUtilArg]) of
+        %% return 0.0 when `emqx_cpu_sup_worker` is not started
```
Comment (thalesmg) [inline apps/emqx_management/src/emqx_mgmt.erl:214]: Misplaced comment?

File: changes/ce/fix-12663.en.md [hunk 3]
```diff
@@ -0,0 +1 @@
+Fixed an issue where the sample `emqx_vm_cpu_use` and `emqx_vm_cpu_idle` in prometheus endpoint `/prometheus/stats` are always the values since system bootup.
```
Comment (thalesmg) [inline changes/ce/fix-12663.en.md:1]:
```suggestion
Fixed an issue where `emqx_vm_cpu_use` and `emqx_vm_cpu_idle` metrics in Prometheus endpoint `/prometheus/stats` are unchanged after system boot.
```

File: apps/emqx/src/emqx_cpu_sup_worker.erl [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+handle_call(cpu_util, _From, State) ->
+    Val = cpu_sup:util(),
+    {reply, Val, State#{cpu_util => Val}};
+handle_call({cpu_util, Args}, _From, State) ->
+    Val = erlang:apply(cpu_sup, util, Args),
+    {reply, Val, State};
+handle_call(Req, _From, State) ->
+    {reply, {error, {unexpected_call, Req}}, State}.
+
+handle_cast(Msg, State) ->
```
Comment (thalesmg) [inline apps/emqx/src/emqx_cpu_sup_worker.erl:89]: Shouldn't this handle the `monitor_conf_update` cast from l. 65?

File: changes/ce/fix-12663.en.md (hunk 3 above)
Comment (JimMoen) [reply changes/ce/fix-12663.en.md:1]: Strictly speaking, the CPU usage is not "unchanged after system boot."
It's just that every time the usage is obtained, the calculated usage is the average usage "from system startup until the API is called."
For example:
8 hours after booting, the average usage during these 8 hours will be calculated.
One day after power-on, the average usage for that day will be calculated.
And the length of time has nothing to do with EMQX startup time.

File: apps/emqx/src/emqx_cpu_sup_worker.erl (hunk 4 above)
Comment (JimMoen) [reply apps/emqx/src/emqx_cpu_sup_worker.erl:89]: nice catch.
It’s useless code that I forgot to delete.

File: apps/emqx/src/emqx_sys_sup.erl (hunk 1 above)
Comment (JimMoen) [reply apps/emqx/src/emqx_sys_sup.erl]: directly call `cpu_sup:util/0` by `emqx_os_mon`.
There is no dependency between them now.
It's safe to use the `one_for_one` strategy within the current supervisor.

File: apps/emqx/src/emqx_os_mon.erl [hunk 5]
```diff
@@ -131,7 +132,7 @@ handle_info({timeout, _Timer, mem_check}, #{sysmem_high_watermark := HWM} = Stat
 handle_info({timeout, _Timer, cpu_check}, State) ->
     CPUHighWatermark = emqx:get_config([sysmon, os, cpu_high_watermark]) * 100,
     CPULowWatermark = emqx:get_config([sysmon, os, cpu_low_watermark]) * 100,
-    CPUVal = emqx_vm:cpu_util(),
+    CPUVal = cpu_sup:util(),
```
Comment (qzhuyan) [inline apps/emqx/src/emqx_os_mon.erl:135]: why this change? for accuracy?

**Review by qzhuyan — APPROVED** [review]

File: apps/emqx/src/emqx_os_mon.erl (hunk 5 above)
Comment (JimMoen) [reply apps/emqx/src/emqx_os_mon.erl:135]: `cpu_sup:util/0` will be called in single gen_server `emqx_os_mon`, no need to warp it in `emqx_cpu_sup_worker` process.

**Review by thalesmg — APPROVED** [review]
